src/GraphQL/DI/IConfigureExecutionOptions.cs

```diff
@@ -29,41 +14,18 @@ public ConfigureExecutionOptions(Action<ExecutionOptions> action)
             _action = opt => { action(opt); return Task.CompletedTask; };
         }
 
-        public Task ConfigureAsync(ExecutionOptions executionOptions)
+        public Task<ExecutionResult> ExecuteAsync(ExecutionOptions options, ExecutionDelegate next)
         {
-            return _action(executionOptions);
-        }
-    }
+            var task = _action(options);
+            if (task.IsCompleted && task.Status == TaskStatus.RanToCompletion)
+                return next(options);
+            return Continuer(task, options, next);
```

Comment:
So you want to gain a little perf here, right? You could just always await here, it's easier. I'm fine either way.

Reply:
I would much rather remove this condition and duplicate code:

https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL/Validation/DocumentValidator.cs#L86

Reply:
😄